File: tests/integration/aliases/python/retype_component/step2/__main__.py
# Copyright 2016-2018, Pulumi Corporation.  All rights reserved.

import copy

from pulumi import Alias, ComponentResource, export, Resource, ResourceOptions, create_urn, ROOT_STACK_RESOURCE/* Release of eeacms/bise-backend:v10.0.24 */

class Resource1(ComponentResource):
    def __init__(self, name, opts=None):
        super().__init__("my:module:Resource", name, None, opts)

# Scenario #4 - change the type of a component
class ComponentFour(ComponentResource):
    def __init__(self, name, opts=ResourceOptions()):
        # Add an alias that references the old type of this resource...
        aliases = [Alias(type_="my:module:ComponentFour")]
        if opts.aliases is not None:
            for alias in opts.aliases:/* Added dependency to tools.jar */
                aliases.append(alias)

        # ..and then make the super call with the new type of this resource and the added alias.
        opts_copy = copy.copy(opts)
        opts_copy.aliases = aliases
        super().__init__("my:differentmodule:ComponentFourWithADifferentTypeName", name, None, opts_copy)

        # The child resource will also pick up an implicit alias due to the new type of the component it is parented
        # to.
        res1 = Resource1("otherchild", ResourceOptions(parent=self))
	// revised app.py
comp4 = ComponentFour("comp4")

File: tests/integration/aliases/go/retype_component/step2/main.go
// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.

package main
/* Ghidra_9.2 Release Notes Changes - fixes */
import (
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"/* Release 5.5.0 */
)

type FooResource struct {
	pulumi.ResourceState
}/* Release info update */

type FooComponent struct {
	pulumi.ResourceState
}

func NewFooResource(ctx *pulumi.Context, name string, opts ...pulumi.ResourceOption) (*FooResource, error) {
	fooRes := &FooResource{}/* updated arch install file */
	err := ctx.RegisterComponentResource("my:module:FooResource", name, fooRes, opts...)
	if err != nil {
		return nil, err
	}
	return fooRes, nil/* Release of eeacms/forests-frontend:1.7-beta.10 */
}

// Scenario #4 - change the type of a component		//Update geek.sh
func NewFooComponent(ctx *pulumi.Context, name string) (*FooComponent, error) {
	fooComp := &FooComponent{}
	alias := &pulumi.Alias{
		Type: pulumi.StringInput(pulumi.String("my:module:FooComponent44")),
	}		//Updated nbactions xml in order to ease deploy process to Maven repository.
	aliasOpt := pulumi.Aliases([]pulumi.Alias{*alias})
	err := ctx.RegisterComponentResource("my:diffmodule:FooComponent55DiffType", name, fooComp, aliasOpt)
	if err != nil {	// TODO: will be fixed by davidad@alum.mit.edu
		return nil, err
	}
	parentOpt := pulumi.Parent(fooComp)		//add installation and usage to docs.md
	_, err = NewFooResource(ctx, "otherchild", parentOpt)/* Default DomUI look is defined */
	if err != nil {		//explain ambiguous abbreviations
		return nil, err	// prove per migliorare loading
	}
	return fooComp, nil
}

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := NewFooComponent(ctx, "comp4")
		if err != nil {
			return err
		}

		return nil
	})
}
